common/cmake/devices.cmake

```diff
@@ -1233,7 +1261,30 @@ function(ADD_FPGA_TARGET)
     add_output_to_fpga_target(${NAME} IO_PLACE ${OUT_IO_REL})
     append_file_dependency(VPR_DEPS ${OUT_IO_REL})
 
-    set(FIX_PINS_ARG --fix_pins ${OUT_IO})
+    if(NOT ${NO_PLACE_CONSTR})
+      append_file_dependency(CONSTR_DEPS ${OUT_IO_REL})
+
+      # Generate LOC constrains
+      string(CONFIGURE ${PLACE_CONSTR_TOOL_CMD} PLACE_CONSTR_TOOL_CMD_FOR_TARGET)
+      separate_arguments(
+        PLACE_CONSTR_TOOL_CMD_FOR_TARGET_LIST UNIX_COMMAND ${PLACE_CONSTR_TOOL_CMD_FOR_TARGET}
+      )
+
+      add_custom_command(
+        OUTPUT ${OUT_CONSTR}
+        DEPENDS ${CONSTR_DEPS}
+        COMMAND ${PLACE_CONSTR_TOOL_CMD_FOR_TARGET_LIST} --input ${OUT_IO} --output ${OUT_CONSTR}
```

Comment:
Rather than requiring the place constraint tool to take the `--input` and `--output` arguments, simply have them reference `OUT_IO` and `OUT_CONSTR`, which is how the other tools work.  Alternatively, use stdin/stdout redirection to pipe OUT_IO to to the tool, and pipe OUT_CONSTR from the tool.

Reply:
Either way, update the DEFINE_ARCH with how the interface works.

Reply:
Done